### PR TITLE
Enhance NetworkPolicy, SecurityPolicy and LBService UTs and add license file header

### PR DESCRIPTION
**Pull request by timdengyun**
This patch is to
1. Add more UTS for NetworkPolicy, SecurityPolicy and LBService controller.
2. Add more UTS for SecurityPolicy service.
3. Add license file header.
4. Change log info format.

### CODE REVIEW

File: pkg/controllers/networkpolicy/networkpolicy_controller.go [hunk 1]
```diff
@@ -88,19 +90,23 @@ func cleanNetworkPolicyErrorAnnotation(ctx context.Context, networkPolicy *netwo
...
+	defer func() {
+		log.Info("Finished reconciling NetworkPolicy", "networkpolicy", req.NamespacedName, "duration(ms)", time.Since(startTime).Milliseconds())
+	}()
+
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResType)
 
 	if err := r.Client.Get(ctx, req.NamespacedName, networkPolicy); err != nil {
-		// IgnoreNotFound returns nil on NotFound errors.
-		if client.IgnoreNotFound(err) == nil {
+		if apierrors.IsNotFound(err) {
```
Comment (timdengyun) [inline pkg/controllers/networkpolicy/networkpolicy_controller.go:109]: @TaoZou1 please help confirm this is correct change in the code.

File: pkg/controllers/securitypolicy/securitypolicy_controller.go [hunk 2]
```diff
@@ -137,12 +139,16 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
+	defer func() {
+		log.Info("Finished reconciling SecurityPolicy CR", "securitypolicy", req.NamespacedName, "duration(ms)", time.Since(startTime).Milliseconds())
+	}()
+
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResType)
 
 	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		// IgnoreNotFound returns nil on NotFound errors.
-		if client.IgnoreNotFound(err) == nil {
+		if apierrors.IsNotFound(err) {
```
Comment (timdengyun) [inline pkg/controllers/securitypolicy/securitypolicy_controller.go:151]: @TaoZou1 please help confirm this is correct change in the code.

**Review by zhengxiexie — APPROVED** [review]
LGTM

File: pkg/controllers/securitypolicy/securitypolicy_controller.go (hunk 2 above)
Comment (TaoZou1) [reply pkg/controllers/securitypolicy/securitypolicy_controller.go:151]: correct

Comment (timdengyun) [reply pkg/controllers/securitypolicy/securitypolicy_controller.go:151]: Thanks for confirmation.

File: pkg/controllers/securitypolicy/securitypolicy_controller.go [hunk 3]
```diff
@@ -69,6 +65,12 @@ func k8sClient(mgr ctrl.Manager) client.Client {
...
 }
 
+func updateFail(r *SecurityPolicyReconciler, c context.Context, o *v1alpha1.SecurityPolicy, e *error) {
+	r.setSecurityPolicyReadyStatusFalse(c, o, metav1.Now(), e)
+	r.Recorder.Event(o, v1.EventTypeWarning, common.ReasonFailUpdate, fmt.Sprintf("%v", *e))
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, MetricResType)
+}
+
 func deleteFail(r *SecurityPolicyReconciler, c context.Context, o *v1alpha1.SecurityPolicy, e *error) {
 	r.setSecurityPolicyReadyStatusFalse(c, o, metav1.Now(), e)
```
[the remark this answers is not in the document]
Comment (timdengyun) [reply pkg/controllers/securitypolicy/securitypolicy_controller.go:75]: Yes, normally, there is no need for `deleteFail` in the controller, but for SecurityPolicy controller, there is a T1 upgrade case, the old CR has added finalizer, in upgrade case, the old CR has to remove finalizer before deletion, So, we keep this CR status update here. 

https://github.com/vmware-tanzu/nsx-operator/pull/835/files#diff-f45f721ba06ab477de36df3522bb24718a5f69d5747f3d6b5ae543fe98d68877R228

For T1, the new created CR, we will not add finalizer any more.

**Review by TaoZou1 — APPROVED** [review]

**Review by zhengxiexie — APPROVED** [review]
LGTM

**Pull request merged**